Kluwer Arbitration.js

```diff
@@ -0,0 +1,1011 @@
+	"label": "Kluwer Arbitration",
+	"creator": "Jonas Zaugg",
+	"target": "^https?://(www\\.|arbitrationblog\\.)?kluwerarbitration\\.com/",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-10-20 12:55:15"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Jonas Zaugg
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+/*
+See test cases for examples. URLs used:
+- Document URL: https://www.kluwerarbitration.com/document/
+- Publication details (whole book): https://www.kluwerarbitration.com/api/publicationdetail?documentId=
+- Document details: https://www.kluwerarbitration.com/api/Document?id=
+- PDF request: https://www.kluwerarbitration.com/document/print?title=PDF&ids=
+*/
+
+function getType(pubDetails, docDetails) {
+	let pubType = pubDetails.publicationType ? pubDetails.publicationType : docDetails.PublicationType;
+	let hasDocument = docDetails !== null;
+
+	switch (pubType) {
+		case 'book':
+			// Multiple here would show a selection screen with the option to save the book or individual chapters, but we use the selection screen to indicate which items to download as part of the book
+			return hasDocument ? 'bookSection' : 'book';
+		case 'journal':
+			switch (docDetails.Type) {
+				case 'Court Decisions':
+				case 'Awards':
+					return 'case';
+				case 'Conventions':
+				case 'Rules':
+				case 'Legislation':
+				case 'BITs':
+					return 'statute';
+				case 'Commentary':
+					return 'journalArticle';
+			}
+			break;
+		case 'loose-leaf':
+			switch (docDetails.Type) {
+				case 'Models':
+					return 'statute';
+				case 'Commentary':
+					return 'report';
+			}
+			break;
+		case 'internet':
+			switch (docDetails.Type) {
+				case 'Awards':
+					return 'case';
+				default:
+					return 'webpage';
+			}
+		case 'blog': // Only if we're reading from the search results JSON
+			return 'blogPost';
+	}
+
+	Z.debug("Could not match pairing: " + pubType + ", " + docDetails.Type);
+	return false;
+}
+
+// Function used to get all documents contained within a collection (such as chapters of a book)
+function flattenNodes(tocNodes) {
+	let result = [];
+	recursion(0, tocNodes, result);
+	return result;
+}
+
+function recursion(index, inputArray, outputArray) {
```

Comment:
Will definitely rename and add comments. As a primer, book chapters or articles are treated as "documents" whereas journal issues and books are "publications". The table of contents that's shown for navigations purposes on the website is part of the "publication". The translator uses the ToC either to get the page numbers for an article or individual contribution in a book, or to get all chapters if what we're adding is the book itself (and uses `selectItems` to select the chapters to download).
Because that ToC can have an arbitrary depth and not all nodes have a document (i.e., some are just "Part I", "Part II", etc.), we need to flatten the structure and filter it to get all documents contained within that same publication in a single array for the aforementioned purposes. I don't think there's an alternative here except recursion, but to clarify regarding performance, the publication-level ToC nodes don't go deeper than the document-level. In other words, while there might be some depth due to chapter grouping, you will only go as deep as the chapter/article level, so there should be 10-30 nodes in the end.